libr/anal/fcn.c

```diff
@@ -503,7 +503,9 @@ static int fcn_recurse(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut64 len, int
 				fcn_takeover_block_recursive (fcn, existing_bb);
 			}
 		}
-		r_anal_block_unref (existing_bb);
+		if (existing_bb) {
```

Comment:
wtf? imho r_anal_block_unref should work like free(). accepting NULL as arg

Reply:
No. unref is not free, if you call unref and don't exactly know if it's NULL or not, you are doing something horribly wrong because a unref must always follow a corresponding ref.